ast/compile.go

```diff
@@ -1452,7 +1452,7 @@ func rewritePrintCalls(gen *localVarGenerator, getArity func(Ref) int, globals V
 	// Visit comprehension bodies recursively to ensure print statements inside
 	// those bodies only close over variables that are safe.
 	for i := range body {
-		if ContainsComprehensions(body[i]) {
+		if ContainsComprehensions(body[i]) || body[i].IsEvery() {
```

Comment:
nit: What about adding a new "ContainsClosure" function that we can call instead to deal with comprehensions and every?

Reply:
✔️ Good idea.